### PR TITLE
chore: move team mentions to ce

### PR DESCRIPTION
**Pull request by heitortanoue**
## Proposed changes (including videos or screenshots)
Move the Team Mentions feature to the RC Community Edition

## Steps to test or reproduce

Without the EE license, try to use the team mention feature by sending a message with `@team`, all team members should be notified.

## Further comments

TC-901

### CODE REVIEW

**Review by yash-rajpal — DISMISSED** [review]
LGTM

File: apps/meteor/app/mentions/server/getMentionedTeamMembers.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+callbacks.add('beforeGetMentions', async (mentionIds: string[], extra?: IExtraDataForNotification) => {
+	const { otherMentions } = extra ?? {};
+
+	const teamIds = otherMentions?.filter(({ type }) => type === 'team').map(({ _id }) => _id);
+
+	if (!teamIds?.length) {
+		return mentionIds;
+	}
+
+	const members: ITeamMember[] = await Team.getMembersByTeamIds(teamIds, { projection: { userId: 1 } });
```
Comment (KevLehman) [inline apps/meteor/app/mentions/server/getMentionedTeamMembers.ts:21]: Don't you get the type from getMembersByTeamIds?

File: apps/meteor/app/mentions/server/getMentionedTeamMembers.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	mentionIds.push(
+		...new Set(members.map(({ userId }: { userId: string }) => userId).filter((userId: string) => !mentionIds.includes(userId))),
```
Comment (KevLehman) [inline apps/meteor/app/mentions/server/getMentionedTeamMembers.ts:23]: `members` already has a type, so it should be inferred here 👀

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]
I guess we should also remove `'teams-mention'` from `BundleFeature` and from the enterprise bundle (in `bundles.ts`), shouldn't we? :thinking:

**Review by matheusbsilva137 — APPROVED** [review]

**Pull request merged**